### PR TITLE
Web fix

### LINKED ISSUE #3: Form for sportresults
Create a from for the input of the sportresults of each student.

### PR DESCRIPTION
**Pull request by skyliam**
Close #3

### CODE REVIEW

File: web/js/students_of_class.js [hunk 1]
```diff
@@ -114,18 +176,20 @@ function loadEditSportResult(sportResult) {
     $("div.form-group select").removeAttr('selected')
         .find('option[value=' + discipline + ']')
         .attr('selected', true);
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]:
```javascript
const disciplineSelect = document.querySelector("#discipline");
disciplineSelect.value = discipline;
```

This is equivalent to your code.

File: web/js/students_of_class.js [hunk 2]
```diff
@@ -191,18 +252,14 @@ function constructStudentTableRow(student) {
...
-
     let gender = document.createElement("td");
     gender.innerText = student.female ? "Weiblich" : "Männlich";
     row.appendChild(gender);
 
     let edit = document.createElement("td");
     let buttonEdit = document.createElement("span");
     buttonEdit.onclick = () => {
-        document.getElementById("studentURL").value = studentURL;
+        document.querySelector('#editStudentForm > input[name="studentURL"]').value = studentURL;
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: This line would probably fit better in the `loadEditModal()`-function

File: web/js/students_of_class.js [hunk 3]
```diff
@@ -215,7 +272,7 @@ function constructStudentTableRow(student) {
     let remove = document.createElement("td");
     let buttonRemove = document.createElement("span");
     buttonRemove.onclick = () => {
-        document.getElementById("studentURL").value = studentURL;
+        document.querySelector("#studentURL").value = studentURL;
         modalDeletion.modal('show');
         return false;
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: I don't get the purpose of this line ...

File: web/js/students_of_class.js [hunk 4]
```diff
@@ -29,16 +30,41 @@ const modalEditStudent = $('#editStudentModal').modal({
...
+    clearStudentSportResultTable();
+    showTableColumns("isNotRun");
+});
+
 function createSportResultTable(studentURL, student) {
-    const labelScore = document.getElementById("labelScore");
+    const labelScore = document.querySelector("#labelScore");
     labelScore.innerText = "Punkte: " + student.score;
 
     const SportResultsTableBody = document.querySelector("#sportResults-tbody");
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: Please use CamelCase 😆

File: web/js/students_of_class.js [hunk 5]
```diff
@@ -230,7 +287,7 @@ function constructStudentTableRow(student) {
     let addSportResult = document.createElement("td");
     let addSportResultButton = document.createElement("span");
     addSportResultButton.onclick = () => {
-        document.getElementById("studentURL").value = studentURL;
+        document.querySelector("#studentURL").value = studentURL;
         createSportResultTable(studentURL, student);
         modalSportResult.modal('show');
         return false;
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: Why does this onClick return anything ... ❓

File: web/js/students_of_class.js [hunk 6]
```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
     return row;
 }
 
-function deleteStudentRequest() {
+async function deleteStudentRequest() {
     modalDeletion.modal('hide');
     const errorElement = document.querySelector("#error");
-    const student = document.getElementById("studentURL").value;
-    deleteStudent(student)
+    const student = document.querySelector("#studentURL").value;
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: I think this doesn't access the right input field there is only one in the edit student modal

File: web/js/students_of_class.js [hunk 7]
```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
...
+    await fetchApi();
 }
 
 function updateSchoolClass(schoolClass) {
-    const className = document.getElementById("class-name");
+    const className = document.querySelector("#class-name");
     className.innerHTML = `${schoolClass.grade}${schoolClass.className}`;
 
-    const classTitle = document.getElementById("title");
+    const classTitle = document.querySelector("#title");
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: Please rename this to `websiteTitle` or anything similar. This is quite confusing ...

File: web/js/students_of_class.js [hunk 8]
```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
...
+            const discipline = document.querySelector("#disciplines").value;
+            const sportResult = {result: result, discipline: discipline, student: student.innerText};
+            postSportResult(sportResult)
+                .catch(() => {
+                    errorElement.innerHTML = "The post request was not successful.";
+                    $(errorElement).slideDown().delay(3000).slideUp();
+                })
+        }
+    });
+}
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: I think we should refactor this method together, this is quite complex.

File: web/js/students_of_class.js [hunk 9]
```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
...
+    const classTeacher = document.querySelector("#class-teacher");
     classTeacher.innerHTML = schoolClass.classTeacherName ?? '';
 
-    document.getElementById("classURL").value = schoolClass._links.self.href;
+    document.querySelector("#classURL").value = schoolClass._links.self.href;
+}
+
+function addSportResults() {
+    const table = document.getElementsByClassName("value");
+    const students = document.getElementsByClassName("student");
```
Comment (aykborstelmann) [inline web/js/students_of_class.js]: If you would use `document.querySelectorAll(".student")` you would save the `Array.from`.

File: web/js/students_of_class.js [hunk 10]
```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
...
-    fetchApi();
+saveSportResultsButton.addEventListener('click', function () {
+    addSportResults();
 });
 
+remove.addEventListener('click', deleteStudentRequest, true);
+
+finish.addEventListener('click', function () {
+
+});
```
[the remark this answers is not in the document]
Comment (skyliam) [reply web/js/students_of_class.js]: yes its relevant

File: web/js/students_of_class.js [hunk 11]
```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
...
-    const classBreadCrumb = document.getElementById("breadCrumb");
+    const classBreadCrumb = document.querySelector("#breadCrumb");
     classBreadCrumb.innerHTML = `${schoolClass.grade}${schoolClass.className}`;
 
-    const classTeacher = document.getElementById("class-teacher");
+    const classTeacher = document.querySelector("#class-teacher");
     classTeacher.innerHTML = schoolClass.classTeacherName ?? '';
 
-    document.getElementById("classURL").value = schoolClass._links.self.href;
+    document.querySelector("#classURL").value = schoolClass._links.self.href;
```
[the remark this answers is not in the document]
Comment (skyliam) [reply web/js/students_of_class.js]: I already fixed the issue

File: web/js/students_of_class.js [hunk 12]
```diff
@@ -29,16 +30,41 @@ const modalEditStudent = $('#editStudentModal').modal({
...
+        hideTableColumns("isNotRun");
+    } else {
+        showTableColumns("isNotRun");
+    }
+});
+
+$('#addSportResultsModal').on('show.bs.modal', function () {
+    const studentSportResultsTableBody = document.querySelector("#studentSportResult-tbody");
+    const urlSearchParams = new URLSearchParams(window.location.search);
+    const schoolClassUrl = urlSearchParams.get("schoolClass");
```
[the remark this answers is not in the document]
Comment (skyliam) [reply web/js/students_of_class.js]: This is a good idea

File: web/js/students_of_class.js (hunk 5 above)
Comment (skyliam) [reply web/js/students_of_class.js]: I dont know

File: web/js/students_of_class.js (hunk 8 above)
Comment (skyliam) [reply web/js/students_of_class.js]: Alright

**Comment by aykborstelmann** [comment]
:thumbs_up:

**Pull request merged**